### PR TITLE
HDDS-10036. OzoneManagerRatisServer.getServer() should return Division.

### PR DESCRIPTION
**Pull request by szetszwo**
## What changes were proposed in this pull request?

`OzoneManagerRatisServer` has a specific raftGroup. The `getServer()` method should return the division specific to that group. Otherwise, callers have to get groupId and then get the division with try-catch.

## What is the link to the Apache JIRA

HDDS-10036

## How was this patch tested?

By updating existing tests.

### CODE REVIEW

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOzoneManagerPrepare.java [hunk 1]
```diff
@@ -338,11 +339,9 @@ public void testCancelPrepare() throws Exception {
...
 
   private boolean logFilesPresentInRatisPeer(OzoneManager om) {
-    String ratisDir = om.getOmRatisServer().getServer().getProperties()
-        .get("raft.server.storage.dir");
-    String groupIdDirName =
-        om.getOmRatisServer().getServer().getGroupIds().iterator()
-            .next().getUuid().toString();
+    final RaftServer.Division server = om.getOmRatisServer().getServerDivision();
+    final String ratisDir = server.getRaftServer().getProperties().get("raft.server.storage.dir");
+    final String groupIdDirName = server.getGroup().getGroupId().toString();
```
Comment (adoroszlai) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOzoneManagerPrepare.java:344]: `TestOzoneManagerPrepare` is failing due to mismatch in the directory name.

```suggestion
    final String groupIdDirName = server.getGroup().getGroupId().getUuid().toString();
```

Comment (szetszwo) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOzoneManagerPrepare.java:344]: That is a good catch!  Thanks for the tip.

**Review by adoroszlai — APPROVED** [review]
Thanks @szetszwo for the improvement.

**Comment by szetszwo** [comment]
@adoroszlai , thanks a lot for reviewing and merging this!

**Pull request merged**